piper/ratbagd.py

```diff
@@ -343,14 +364,16 @@ def get_svg(self, theme):
         """
         return self._dbus_call("GetSvg", "s", theme)
 
-    def commit(self):
-        """Commits all changes made to the device."""
-        ret = self._dbus_call("Commit", "")
+    def commit(self, callback):
+        """Commits all changes made to the device.
+
+        @param callback The function to call with the result of the commit
+        """
+        self._dbus_call_async("Commit", "", callback=callback)
         for profile in self._profiles:
```

Comment:
Shouldn't this also be done only if the commit succeeded? As it is right now, we might unflag a dirty profile while the commit fails in the background.

Reply:
not with the plan of assuming commit always succeeds, see https://github.com/libratbag/libratbag/pull/275#discussion_r134724671

if commit fails, then we need to resync *everything* anyway and then we can also assume nothing is dirty.

Reply:
In that case, can we add a comment saying as such?

Also, I assume the resyncing will happen from within libratbag/ratbagd and that these changes will be signaled over the bus so that Piper picks that up automatically?

Reply:
> In that case, can we add a comment saying as such?

done

> Also, I assume the resyncing will happen from within libratbag/ratbagd and that these changes will be signaled over the bus so that Piper picks that up automatically?

yep, at least that's what ratbagd is supposed to do. if it doesn't, it's a bug there